### DISCUSSION TITLE
Documentation for extensions

### DISCUSSION THREAD

**Discussion by nickrum** [directus/directus#3050] [Feature Requests]
I think it would be very helpful if every extension would include it's own documentation. This could then be shown dynamically inside the documentation module based on the installed extensions. This would use the integrated documentation module to its full potential.

Since to my knowledge extensions always consist of a single file, there are two options to approach this:

1. Bundle the documentation as a string in the single javascript file.
2. Use some kind of manifest for extensions, which points to the documentation file.

Option 1 would be the easier one to implement, but is kind of ugly and the filesize of the extension would very quickly blow up.
Option 2 would require changes to the extension architecture, but would be much cleaner. Additionally an extension manifest could be used for other things like making the API aware of the name, icon and path of a custom module, sending that to the App and only loading the module inside the App if it is actually used.

What do you think of this idea?

**Comment by rijkvanzanten** [comment]
I think it's a great idea. This combined with the idea to be able to plug in your own custom docs as an extension will make the docs module crazy powerful.

A third option would be to use a standardized file name for these files. We already rely on the entry point file to be called `index.js`, so we could say that a `docs.md` file will be read in and used as is

**Reply by nickrum** [reply]
That's awesome. I like the idea of custom docs extensions as well.

I actually thought of that option as well but must have forgotten about it since then. If the entry point already relies on a standard then this would be the most obvious option.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
